src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java

```diff
@@ -1386,10 +1387,8 @@ public ValidationResult validate( StyleInfo style, boolean isNew ) {
     
     public void remove(StyleInfo style) {
         //ensure no references to the style
-        for ( LayerInfo l : facade.getLayers() ) {
-            if ( style.equals( l.getDefaultStyle() ) || l.getStyles().contains( style )) {
-                throw new IllegalArgumentException( "Unable to delete style referenced by '"+ l.getName()+"'");
-            }
+        for ( LayerInfo l : facade.getLayers(style) ) {
+            throw new IllegalArgumentException( "Unable to delete style referenced by '"+ l.getName()+"'");
```

Comment:
I actually don't know, the code was already doing that, I just removed a redundant if in there.